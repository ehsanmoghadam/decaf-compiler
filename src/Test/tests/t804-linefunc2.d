void func__line__func() {
    int line;
    string func;
    int input;
    string out;

    func = __func__;
    line = __line__;
    input = ReadInteger();
    Print("Function name: ", func);
    Print("Line number: ", line);
    if ((input + line) % 2 == 0) {
        out = "Line number plus input is even!";
    } else {
        out = "Line number plus input is odd!";
    }
    Print(out);
}
