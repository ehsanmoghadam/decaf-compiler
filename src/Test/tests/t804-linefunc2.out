Function name: func__line__func
Line number: 8
Line number plus input is odd!
